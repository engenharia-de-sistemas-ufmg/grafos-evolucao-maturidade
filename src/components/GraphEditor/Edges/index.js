import { connect } from 'react-redux';
import EdgesEditor from './EdgesEditor';
import { updateEdges } from '../../../actions/graph';

const mapStateToProps = (state) => ({
  lang: state.page.lang,
  edges: state.graph.edges,
});

const mapDispatchToProps = (dispatch) => ({
  updateEdges: (edges) => { dispatch(updateEdges(edges)); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(EdgesEditor);
