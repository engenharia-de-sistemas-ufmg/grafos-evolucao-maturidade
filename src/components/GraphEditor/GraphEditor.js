/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import * as labels from '../../commons/labels';
import NodesEditor from './Nodes/index';
import NodeGroupsEditor from './NodeGroups/index';
import EdgesEditor from './Edges/index';
import OptionsEditor from './Options/index';

const GraphEditor = ({
  lang, pageMode,
  editNodes, editNodeGroups, editEdges, editOptions,
}) => (
  <div className="box">
    <div className="tabs is-small">
      <ul>
        <li className={pageMode === 'editing_nodes' ? 'is-active' : ''}>
          <a onClick={editNodes}>{labels.nodes[lang]}</a>
        </li>
        <li className={pageMode === 'editing_node_groups' ? 'is-active' : ''}>
          <a onClick={editNodeGroups}>{labels.nodeGroups[lang]}</a>
        </li>
        <li className={pageMode === 'editing_edges' ? 'is-active' : ''}>
          <a onClick={editEdges}>{labels.edges[lang]}</a>
        </li>
        <li className={pageMode === 'editing_options' ? 'is-active' : ''}>
          <a onClick={editOptions}>{labels.options[lang]}</a>
        </li>
      </ul>
    </div>
    <div>
      {pageMode === 'editing_nodes' && <NodesEditor />}
      {pageMode === 'editing_node_groups' && <NodeGroupsEditor />}
      {pageMode === 'editing_edges' && <EdgesEditor />}
      {pageMode === 'editing_options' && <OptionsEditor />}
    </div>
  </div>
);

GraphEditor.propTypes = {
  lang: PropTypes.string.isRequired,
  pageMode: PropTypes.string.isRequired,
  editNodes: PropTypes.func.isRequired,
  editNodeGroups: PropTypes.func.isRequired,
  editEdges: PropTypes.func.isRequired,
  editOptions: PropTypes.func.isRequired,
};

export default GraphEditor;
