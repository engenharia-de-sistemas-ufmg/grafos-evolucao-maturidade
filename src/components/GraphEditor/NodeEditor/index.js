import { connect } from 'react-redux';
import NodeEditor from './NodeEditor';
import { endNodeEdition } from '../../../actions/index';

const mapStateToProps = (state) => ({
  nodeEdition: state.page.nodeEdition,
});

const mapDispatchToProps = (dispatch) => ({
  endNodeEdition: () => { dispatch(endNodeEdition()); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NodeEditor);
