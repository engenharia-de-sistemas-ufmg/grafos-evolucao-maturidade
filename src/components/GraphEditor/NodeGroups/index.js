import { connect } from 'react-redux';
import NodeGroupsEditor from './NodeGroupsEditor';
import { updateNodeGroups } from '../../../actions/index';

const mapStateToProps = (state) => ({
  lang: state.page.lang,
  nodeGroups: state.nodeGroups,
});

const mapDispatchToProps = (dispatch) => ({
  updateNodeGroups: (nodeGroups) => { dispatch(updateNodeGroups(nodeGroups)); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NodeGroupsEditor);
