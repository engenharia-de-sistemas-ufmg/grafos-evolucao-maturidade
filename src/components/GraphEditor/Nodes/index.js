import { connect } from 'react-redux';
import NodesEditor from './NodesEditor';
import { updateNodes } from '../../../redux/graph.actions';

const mapStateToProps = (state) => ({
  lang: state.page.lang,
  nodes: state.graph.nodes,
});

const mapDispatchToProps = (dispatch) => ({
  updateNodes: (nodes) => { dispatch(updateNodes(nodes)); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NodesEditor);
