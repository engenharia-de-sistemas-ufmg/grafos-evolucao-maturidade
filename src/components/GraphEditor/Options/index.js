import { connect } from 'react-redux';
import OptionsEditor from './OptionsEditor';
import { updateOptions } from '../../../actions/index';

const mapStateToProps = (state) => ({
  lang: state.page.lang,
  options: state.options,
});

const mapDispatchToProps = (dispatch) => ({
  updateOptions: (options) => { dispatch(updateOptions(options)); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(OptionsEditor);
