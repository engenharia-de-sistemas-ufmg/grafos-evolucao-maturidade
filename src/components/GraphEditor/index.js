import { connect } from 'react-redux';
import GraphEditor from './GraphEditor';
import {
  editNodes, editNodeGroups, editEdges, editOptions,
} from '../../actions/page';

const mapStateToProps = (state) => ({
  pageMode: state.page.mode,
  lang: state.page.lang,
});

const mapDispatchToProps = (dispatch) => ({
  editNodes: () => { dispatch(editNodes()); },
  editNodeGroups: () => { dispatch(editNodeGroups()); },
  editEdges: () => { dispatch(editEdges()); },
  editOptions: () => { dispatch(editOptions()); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(GraphEditor);
