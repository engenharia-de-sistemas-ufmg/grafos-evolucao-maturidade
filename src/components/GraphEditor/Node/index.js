import { connect } from 'react-redux';
import NodeEditor from './NodeEditor';
import { updateNodes } from '../../../actions/graph';

const mapStateToProps = (state) => ({
  pageMode: state.page.mode,
  lang: state.page.lang,
  nodes: state.graph.nodes,
});

const mapDispatchToProps = (dispatch) => ({
  updateNodes: (nodes) => { dispatch(updateNodes(nodes)); },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NodeEditor);
