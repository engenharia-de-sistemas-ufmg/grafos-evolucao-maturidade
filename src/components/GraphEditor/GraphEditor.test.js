import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import GraphEditor from './GraphEditor';

describe('GraphEditor', () => {
  Enzyme.configure({ adapter: new Adapter() });

  test('should match snapshot', () => {
    const component = shallow(
      <GraphEditor
        lang="pt"
        pageMode="editing_nodes"
        editNodes={() => { }}
        editNodeGroups={() => { }}
        editEdges={() => { }}
        editOptions={() => { }}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
