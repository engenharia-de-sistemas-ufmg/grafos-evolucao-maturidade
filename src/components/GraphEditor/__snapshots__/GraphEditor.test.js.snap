// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphEditor should match snapshot 1`] = `
<div
  className="box"
>
  <div
    className="tabs is-small"
  >
    <ul>
      <li
        className="is-active"
      >
        <a
          onClick={[Function]}
        >
          Vértices
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Grupos de Vértices
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Arestas
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Opções
        </a>
      </li>
    </ul>
  </div>
  <div>
    <Connect(NodesEditor) />
  </div>
</div>
`;
